### DISCUSSION TITLE
Using mutation but still getting [vuex] do not mutate...

### DISCUSSION THREAD

**Discussion by fundaxi** [quasarframework/quasar#11080] [General - Components / Directives / etc]
This one has been driving me crazy!

I have a function where I call a mutation and I'm still getting the vuex do not mutate error. I've also tried calling an action that calls the commit but received the same error.

```
// random.js
import store from "src/store"
const $store = store()

function() {
  $store.commit("module/userCommit", payload)
}
```

```
// mutations.js
export function userCommit(state, user)  {
  state.currentUser = user
  // I also tried the line below based on another answer in this forum to no avail
  state.currentUser.splice(0, state.currentUser.length, ...user)
};
```

```
// state.js
export default {
  currentUser: null
  // (Also tried as an empty Array based on the splice method in mutations.
}
```

Help GREATLY appreciated!

**Comment by MilosPaunovic** [comment]
Which exact error are you receiving?

**Reply by fundaxi** [reply]
Sorry about being vague! The full error is below.

Uncaught (in promise) Error: [vuex] do not mutate vuex store state outside mutation handlers.

**Comment by hawkeye64** [comment]
`state.currentUser` is reactive. Assigning `user` to it will lose that reactivity. That's why you get the warning. Use vue's `set` method if it's an object, or splice method (with rest operator) if it's an array.

**Reply by fundaxi** [reply]
Thanks hawkeye! That makes complete sense. However, I'm finding I don't have access to Vue from within the store (not attached to "this" in any form either). I've scoured the web and can't find a solution for Vuex 4. I'm going to continue my search again tomorrow unless I wake up to a good solution. :)

Appreciate the assistance!

**Reply by hawkeye64** [reply]
Sorry, the `set` method was only available in Vue v2 as a means of overcoming IE 11 reactivity limitations and is no longer available in v3.

**Reply by fundaxi** [reply]
I see. So what is the solution for assigning an object to the store with Quasar 2, Vue 3, and Vuex 4? Surely there must be some way to do this.

**Reply by hawkeye64** [reply]
Perhaps iterating with `Object.keys()` might be permissible.

**Reply by fundaxi** [reply]
Thank you for all your assistance!

Unfortunately, I still couldn't get Vuex to work. For now, I have implemented Pinia and so far so good!
